lib/ansible/plugins/filter/core.py

```diff
@@ -292,14 +292,14 @@ def get_encrypted_password(password, hashtype='sha512', salt=None):
             if sys.platform.startswith('darwin'):
                 raise AnsibleFilterError('|password_hash requires the passlib python module to generate password hashes on Mac OS X/Darwin')
             saltstring = "$%s$%s" % (cryptmethod[hashtype], salt)
-            encrypted = crypt.crypt(password, saltstring)
+            encrypted = crypt.crypt(str(password), saltstring)
```

Comment:
This should be `to_text` from ` ansible.module_utils._text` instead of `str` for python3 compatibility.

Reply:
Okay, I've tested this and on python2, it needs to be a byte string.  On python3 it needs to be a text string.  ```str``` technically does that but can traceback on python2 if password contains non-ascii chars.  to_native will do the right things here (at least, unless python3's crypt is changed to take a byte string.  The python3 hashlib functions take byte strings so I would have thought this would take a byte string as well.  Oh well, we'll cross that bridge if they change it).
``` python
encrypted = crypt.crypt(to_native(password, errors='surrogate_or_strict'), saltstring)
```

Reply:
I'd also put in a comment because using to_native() in controller code is generally wrong.  The comment will let future contributors know that to_native() fits the API that we're targetting here:
```
# On python3, the password for crypt must be a text string.
# On python2, the password should be bytes because a text string with non-ascii chars will traceback
encrypted = crypt.crypt(to_native(password, errors='surrogate_or_strict'), saltstring)
```